packages/core/src/state-management/state-manager.ts

```diff
@@ -245,7 +245,7 @@ export class StateManager {
       try {
         await this.executionQ.forStream(state$.id).run(async () => {
           await this._handleTip(state$, anchorCommit)
-          if (state$.tip == anchorCommit) {
+          if (state$.tip.equals(anchorCommit)) {
```

Comment:
The previous code was preventing a _published_ anchor commit from being applied in the failing tests.